modules/finality-verifier/src/lib.rs

```diff
@@ -35,73 +35,66 @@
+use frame_support::{ensure, traits::Get};
 use frame_system::ensure_signed;
 use sp_runtime::traits::Header as HeaderT;
 
 #[cfg(test)]
 mod mock;
 
-/// Header of the bridged chain.
-pub(crate) type BridgedHeader<T> = HeaderOf<<T as Config>::BridgedChain>;
-
-/// The module configuration trait.
-pub trait Config: frame_system::Config {
-	/// The chain we are bridging to here.
-	type BridgedChain: Chain;
-	/// The pallet which we will use as our underlying storage mechanism.
-	type HeaderChain: HeaderChain<<Self::BridgedChain as Chain>::Header>;
-	/// The type through which we will verify that a given header is related to the last
-	/// finalized header in our storage pallet.
-	type AncestryChecker: AncestryChecker<
-		<Self::BridgedChain as Chain>::Header,
-		Vec<<Self::BridgedChain as Chain>::Header>,
-	>;
-	/// The maximum length of headers we can have in a single ancestry proof. This prevents
-	/// unbounded iteration when verifying proofs.
-	type MaxHeadersInSingleProof: Get<u8>;
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as FinalityVerifier {}
-}
-
-decl_error! {
-	pub enum Error for Module<T: Config> {
-		/// The given justification is invalid for the given header.
-		InvalidJustification,
-		/// The given ancestry proof is unable to verify that the child and ancestor headers are
-		/// related.
-		InvalidAncestryProof,
-		/// The authority set from the underlying header chain is invalid.
-		InvalidAuthoritySet,
-		/// Failed to write a header to the underlying header chain.
-		FailedToWriteHeader,
-		/// Failed to write finality proof to the underlying header chain.
-		FailedToWriteFinalityProof,
-		/// The given ancestry proof is too large to be verified in a single transaction.
-		OversizedAncestryProof,
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	/// Header of the bridged chain.
+	pub(crate) type BridgedHeader<T> = HeaderOf<<T as Config>::BridgedChain>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The chain we are bridging to here.
+		type BridgedChain: Chain;
+
+		/// The pallet which we will use as our underlying storage mechanism.
+		type HeaderChain: HeaderChain<<Self::BridgedChain as Chain>::Header>;
+
+		/// The type through which we will verify that a given header is related to the last
+		/// finalized header in our storage pallet.
+		type AncestryChecker: AncestryChecker<
+			<Self::BridgedChain as Chain>::Header,
+			Vec<<Self::BridgedChain as Chain>::Header>,
+		>;
+
+		/// The maximum length of headers we can have in a single ancestry proof. This prevents
+		/// unbounded iteration when verifying proofs.
+		#[pallet::constant]
+		type MaxHeadersInSingleProof: Get<u8>;
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		type Error = Error<T>;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
```

Comment:
Is this even needed? `#[pallet::storage]` itself is optional and according to your own comment, we don't have storage items?

Reply:
Good catch, I don't think it's required